src/controllers/filters/lang/Filter.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "controllers/filters/lang/expressions/Expression.hpp"
+#include "controllers/filters/lang/Types.hpp"
+
+#include <QString>
+
+#include <memory>
+#include <variant>
+
+namespace chatterino {
+
+class Channel;
+struct Message;
+using MessagePtr = std::shared_ptr<const Message>;
+
+}  // namespace chatterino
+
+namespace chatterino::filters {
+
+static const QMap<QString, Type> messageTypingContext = {
+    {"author.badges", Type::StringList},
+    {"author.color", Type::Color},
+    {"author.name", Type::String},
+    {"author.no_color", Type::Bool},
+    {"author.subbed", Type::Bool},
+    {"author.sub_length", Type::Int},
+    {"channel.name", Type::String},
+    {"channel.watching", Type::Bool},
+    {"channel.live", Type::Bool},
+    {"flags.highlighted", Type::Bool},
+    {"flags.points_redeemed", Type::Bool},
+    {"flags.sub_message", Type::Bool},
+    {"flags.system_message", Type::Bool},
+    {"flags.reward_message", Type::Bool},
+    {"flags.first_message", Type::Bool},
+    {"flags.elevated_message", Type::Bool},
+    {"flags.cheer_message", Type::Bool},
+    {"flags.whisper", Type::Bool},
+    {"flags.reply", Type::Bool},
+    {"flags.automod", Type::Bool},
+    {"message.content", Type::String},
+    {"message.length", Type::Int}};
```

Comment:
```suggestion
    {"message.length", Type::Int},
};
```

Comma at the end for easier diffing in the future